spec/rules_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rules'
+
+describe Rules do
+
+  before(:each) do
+    @board = Board.new(3, 3)
+    @current_game =  Rules.new(@board)
+  end
+
+  it "should check for a horizontal win" do
+    @board.grid = [['X', 'X', 'X'],
+                   [nil, nil, nil],
+                   [nil, nil, nil]]
+    expect(@current_game.horizontal_win).to eq('X')
+  end
+
+  it "should check for a vertical win" do
+    @board.grid = [['X', nil, nil],
+                   ['X', nil, nil],
+                   ['X', nil, nil]]
+    expect(@current_game.vertical_win).to eq('X')
+  end
+
+  it "should check for a diagonal win" do
+    @board.grid = [['X', nil, nil],
+                   [nil, 'X', nil],
+                   [nil, nil, 'X']]
+    expect(@current_game.diagonal_win).to eq('X')
+  end
+
+  it "should check for a diagonal win" do
+    @board.grid = [[nil, nil, 'X'],
+                   [nil, 'X', nil],
+                   ['X', nil, nil]]
+    expect(@current_game.diagonal_win).to eq('X')
+  end
+
+  it "should check for a tie" do
+    @board.grid = [['O', 'X', 'X'],
+                   ['X', 'O', 'O'],
+                   ['X', 'O', 'X']]
+    expect(@current_game.tie).to eq(true)
+  end
+
+  it "should check for a tie with empty spaces" do
+    @board.grid = [['O', 'X', 'X'],
+                   ['X', 'O', 'O'],
+                   ['X', 'O', nil]]
+    expect(@current_game.tie).to eq(false)
+  end
+
+  it "should check for a tie without empty spaces" do
+    @board.grid = [['O', 'O', 'X'],
+                   ['X', 'O', 'O'],
+                   ['X', 'O', 'X']]
+    expect(@current_game.tie).to eq(false)
```

Comment:
This should result in `true`. Also, there isn't a real need for this test since it's virtually testing the same behavior as line 38.

Reply:
Yes, you're right. This test does not test anything new. DELETED

Reply:
Oh! I just remembered the reasoning behind the tests related to this 'tie' method. The three scenarios I wanted to test:
(1) when board is full AND no wins ("is_tie" should return true)
(2) when board is not yet completely full AND no wins yet ("is_tie" should return false)
(3) when board is full AND a player has a win ("is_tie" should return false - the board.grid referenced above) 

The assumption I made about what makes a 'tie' is that the board should be full (no spaces with nil values) and have no winner. Since an empty space can potentially still leave an opportunity for a player to win, I made this assumption.